lxd/patches.go

```diff
@@ -181,6 +182,51 @@ func patchesApply(d *Daemon, stage patchStage) error {
 
 // Patches begin here
 
+// patchDBNodesAutoInc re-creates the nodes table id column as AUTOINCREMENT.
+// Its done as a patch rather than a schema update so we can use PRAGMA foreign_keys = OFF without a transaction.
+func patchDBNodesAutoInc(name string, d *Daemon) error {
+	row := d.State().Cluster.DB().QueryRow("SELECT sql FROM sqlite_master WHERE name = 'nodes'")
+
+	var schemaSQL string
+	err := row.Scan(&schemaSQL)
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(schemaSQL, "id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL") {
+		logger.Debugf("Skipping %q patch as node table id column already AUTOINCREMENT", name)
+		return nil // Nothing to do.
+	}
+
+	_, err = d.State().Cluster.DB().Exec(`
+PRAGMA foreign_keys = OFF;
+PRAGMA legacy_alter_table = ON;
```

Comment:
Isn't that global? If so, you definitely need to revert that at the end.

Reply:
Mmm yeah if the connection is held open after patches then yes. I'll also add a check for leader so there's no chance of a race between checking the existing schema and applying the fix if missing.